app/controllers/courses/sites_controller.rb

```diff
@@ -8,6 +8,24 @@ class SitesController < ApplicationController
 
     def edit; end
 
+    def update
+      @course.provider_code = @provider.provider_code
+      @course.sites = params.dig(:course, :site_statuses_attributes)
+        &.values
+        &.select { |field| field["selected"] == "1" }
+        &.map { |field| @provider.sites.find { |site| site.id == field["id"] } }
+
+      if @course.save
+        @course.sync_with_search_and_compare(provider_code: params[:provider_code])
+
+        redirect_to provider_course_path(params[:provider_code], params[:code]), flash: { success: 'Course locations published' }
```

Comment:
We should only say Published if the course is published, otherwise we should say Saved